File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/CalculoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.Periodo;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import br.jus.trt8.pjecalc.base.comum.relatorio.JREmptyDS;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.AbstractResumoPrecatorioJrAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.ApuracaoDeJurosJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.CustaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.DemonstrativoJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.EsocialInssFgtsJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.FGTSJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.HonorarioJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.InssJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.IrpfJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.JustificativaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.MultaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.PensaoAlimenticiaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.PrevidenciaPrivadaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.ResumoJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.SalarioFamiliaJRAdapter;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.SeguroDesempregoJRAdapter;
import java.util.Date;

public abstract class CalculoJRAdapter
extends JRAdapter {
    public abstract boolean getMostrarDemonstrativo();

    public abstract boolean getMostrarResumo();

    public abstract boolean getMostrarResumoPrecatorio();

    public abstract boolean getMostrarDemonstrativoFGTS();

    public abstract boolean getMostrarDemonstrativoINSS();

    public abstract boolean getMostrarDemonstrativoEsocialInssFgts();

    public abstract boolean getMostrarPensaoAlimenticia();

    public abstract boolean getMostrarSeguroDesemprego();

    public abstract boolean getMostrarSalarioFamilia();

    public abstract boolean getMostrarPrevidenciaPrivada();

    public abstract boolean getMostrarApuracaoDeJuros();

    public abstract boolean getMostrarMulta();

    public abstract boolean getMostrarHonorario();

    public abstract boolean getMostrarIrpf();

    public abstract boolean getMostrarCustas();

    public abstract boolean getMostrarComentarios();

    public abstract boolean getMostrarJustificativas();

    public abstract String getNumeroDoProcesso();

    public abstract String getNumeroDoCalculo();

    public abstract String getReclamante();

    public abstract String getReclamado();

    public abstract Periodo getPeriodoDeCalculo();

    public abstract Date getDataDeAjuizamento();

    public abstract Date getDataDaLiquidacao();

    public abstract String getComentarios();

    public abstract DemonstrativoJRAdapter getDemonstrativo();

    public abstract ResumoJRAdapter getResumo();

    public abstract AbstractResumoPrecatorioJrAdapter getResumoPrecatorio();

    public abstract FGTSJRAdapter getDemonstrativoFGTS();

    public abstract InssJRAdapter getDemonstrativoINSS();

    public abstract EsocialInssFgtsJRAdapter getDemonstrativoEsocialInssFgts();

    public abstract PensaoAlimenticiaJRAdapter getPensaoAlimenticia();

    public abstract SeguroDesempregoJRAdapter getSeguroDesemprego();

    public abstract SalarioFamiliaJRAdapter getSalarioFamilia();

    public abstract PrevidenciaPrivadaJRAdapter getPrevidenciaPrivada();

    public abstract ApuracaoDeJurosJRAdapter getApuracaoDeJuros();

    public abstract MultaJRAdapter getMulta();

    public abstract HonorarioJRAdapter getHonorario();

    public abstract IrpfJRAdapter getIrpf();

    public abstract CustaJRAdapter getCustas();

    public abstract JustificativaJRAdapter getJustificativa();

    public abstract JRAdapterDataSource<JREmptyDS> getEmptyDS();
}

File: src/lib/pjecalc/core/dominio/calculo/relatorio/calculo-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — CalculoJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.CalculoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../calculo/relatorio/CalculoJRAdapter.java
 *
 * Adapter raiz do relatório do cálculo. Agrega sub-adapters para cada seção
 * (demonstrativo, resumo, FGTS, INSS, IRPF, juros, multa, honor, custas,
 * pensão, seguro desemprego, salário família, previdência privada).
 *
 * Subclasse concreta: `CalculoJRAdapterPadrao`.
 */
import type { Periodo } from '../../../base/comum/periodo';
import { JRAdapter, JRAdapterDataSource, type JREmptyDS } from './jr-adapter';
import type { AbstractResumoPrecatorioJrAdapter } from './abstract-resumo-precatorio-jr-adapter';
import type { ApuracaoDeJurosJRAdapter } from './apuracao-de-juros-jr-adapter';
import type { CustaJRAdapter } from './custa-jr-adapter';
import type { DemonstrativoJRAdapter } from './demonstrativo-jr-adapter';
import type { EsocialInssFgtsJRAdapter } from './esocial-inss-fgts-jr-adapter';
import type { FGTSJRAdapter } from './fgts-jr-adapter';
import type { HonorarioJRAdapter } from './honorario-jr-adapter';
import type { InssJRAdapter } from './inss-jr-adapter';
import type { IrpfJRAdapter } from './irpf-jr-adapter';
import type { JustificativaJRAdapter } from './justificativa-jr-adapter';
import type { MultaJRAdapter } from './multa-jr-adapter';
import type { PensaoAlimenticiaJRAdapter } from './pensao-alimenticia-jr-adapter';
import type { PrevidenciaPrivadaJRAdapter } from './previdencia-privada-jr-adapter';
import type { ResumoJRAdapter } from './resumo-jr-adapter';
import type { SalarioFamiliaJRAdapter } from './salario-familia-jr-adapter';
import type { SeguroDesempregoJRAdapter } from './seguro-desemprego-jr-adapter';

export abstract class CalculoJRAdapter extends JRAdapter {
  // ─── Flags de seção (qual mostrar) ───
  abstract getMostrarDemonstrativo(): boolean;
  abstract getMostrarResumo(): boolean;
  abstract getMostrarResumoPrecatorio(): boolean;
  abstract getMostrarDemonstrativoFGTS(): boolean;
  abstract getMostrarDemonstrativoINSS(): boolean;
  abstract getMostrarDemonstrativoEsocialInssFgts(): boolean;
  abstract getMostrarPensaoAlimenticia(): boolean;
  abstract getMostrarSeguroDesemprego(): boolean;
  abstract getMostrarSalarioFamilia(): boolean;
  abstract getMostrarPrevidenciaPrivada(): boolean;
  abstract getMostrarApuracaoDeJuros(): boolean;
  abstract getMostrarMulta(): boolean;
  abstract getMostrarHonorario(): boolean;
  abstract getMostrarIrpf(): boolean;
  abstract getMostrarCustas(): boolean;
  abstract getMostrarComentarios(): boolean;
  abstract getMostrarJustificativas(): boolean;

  // ─── Cabeçalho ───
  abstract getNumeroDoProcesso(): string;
  abstract getNumeroDoCalculo(): string;
  abstract getReclamante(): string;
  abstract getReclamado(): string;
  abstract getPeriodoDeCalculo(): Periodo;
  abstract getDataDeAjuizamento(): Date;
  abstract getDataDaLiquidacao(): Date;
  abstract getComentarios(): string;

  // ─── Sub-adapters ───
  abstract getDemonstrativo(): DemonstrativoJRAdapter;
  abstract getResumo(): ResumoJRAdapter;
  abstract getResumoPrecatorio(): AbstractResumoPrecatorioJrAdapter;
  abstract getDemonstrativoFGTS(): FGTSJRAdapter;
  abstract getDemonstrativoINSS(): InssJRAdapter;
  abstract getDemonstrativoEsocialInssFgts(): EsocialInssFgtsJRAdapter;
  abstract getPensaoAlimenticia(): PensaoAlimenticiaJRAdapter;
  abstract getSeguroDesemprego(): SeguroDesempregoJRAdapter;
  abstract getSalarioFamilia(): SalarioFamiliaJRAdapter;
  abstract getPrevidenciaPrivada(): PrevidenciaPrivadaJRAdapter;
  abstract getApuracaoDeJuros(): ApuracaoDeJurosJRAdapter;
  abstract getMulta(): MultaJRAdapter;
  abstract getHonorario(): HonorarioJRAdapter;
  abstract getIrpf(): IrpfJRAdapter;
  abstract getCustas(): CustaJRAdapter;
  abstract getJustificativa(): JustificativaJRAdapter;

  /** getEmptyDS — data source vazia para relatórios sem bean coletivo. */
  abstract getEmptyDS(): JRAdapterDataSource<JRAdapter>;

  static createEmptyDataSource<T extends JRAdapter>(adapter: T): JRAdapterDataSource<T> {
    return new JRAdapterDataSource(adapter, [] as JREmptyDS[]);
  }
}
